src/observable/BoundCallbackObservable.ts

```diff
@@ -138,11 +138,21 @@ function dispatch<T>(state: { source: BoundCallbackObservable<T>, subscriber: Su
   self.add(subject.subscribe(subscriber));
 }
 
-function dispatchNext({ value, subject }) {
+interface DispatchNextArg<T> {
+  subject: AsyncSubject<T>;
+  value: T;
+}
+function dispatchNext<T>(arg: DispatchNextArg<T>) {
+  const { value, subject } = arg;
```

Comment:
This might read cleaner (maybe?) as:

```TypeScript
function dispatchNext<T> ( { value, subject }: DispatchNextArg<T> ) {
 ....
}
```

One of the advantages of destructured function args is readability (but that's pretty subjective).

What do you think?